x-pack/platform/plugins/shared/onechat/server/telemetry/usage_counters.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UsageCollectionSetup, UsageCounter } from '@kbn/usage-collection-plugin/server';
+
+/**
+ * Domain ID for all onechat usage counters
+ */
+export const ONECHAT_USAGE_DOMAIN = 'onechat';
+
+/**
+ * Create usage counter for onechat plugin
+ * @param usageCollection - Usage collection setup contract
+ * @returns Usage counter instance
+ */
+export function createOnechatUsageCounter(
+  usageCollection?: UsageCollectionSetup
+): UsageCounter | undefined {
+  if (!usageCollection) {
+    return undefined;
+  }
+
+  return usageCollection.createUsageCounter(ONECHAT_USAGE_DOMAIN);
+}
+
+/**
+ * Helper to track tool calls with source
+ * @param usageCounter - Usage counter instance
+ * @param source - Tool call source
+ */
+export function trackToolCall(
+  usageCounter: UsageCounter | undefined,
+  source: 'default_agent' | 'custom_agent' | 'mcp' | 'api' | 'a2a'
+): void {
+  if (!usageCounter) return;
+
+  usageCounter.incrementCounter({
+    counterName: `tool_call_${source}`,
```

Comment:
I think we want to prepend some plugin prefix to all our counters to avoid risks of potential conflicts, no? iirc that's the recommended approach?